File: seed.py
# Database Seeder Script
# This script creates sample users for all departments

import asyncio
import sys
sys.path.append('./server/src')

from pymongo import MongoClient
import bcrypt
from datetime import datetime

# MongoDB connection
client = MongoClient('mongodb://localhost:27017/')
db = client['regent_dms']

def hash_password(password):
    """Hash password using bcrypt"""
    salt = bcrypt.gensalt(10)
    return bcrypt.hashpw(password.encode('utf-8'), salt).decode('utf-8')

def create_users():
    """Create sample users for all departments"""
    
    users = [
        {
            'firstName': 'Registry',
            'lastName': 'Admin',
            'email': 'registry@regent.edu',
            'password': hash_password('registry123'),
            'department': 'registry',
            'role': 'admin',
            'isActive': True,
            'createdAt': datetime.utcnow(),
            'updatedAt': datetime.utcnow()
        },
        {
            'firstName': 'Accounts',
            'lastName': 'Admin',
            'email': 'accounts@regent.edu',
            'password': hash_password('accounts123'),
            'department': 'accounts',
            'role': 'admin',
            'isActive': True,
            'createdAt': datetime.utcnow(),
            'updatedAt': datetime.utcnow()
        },
        {
            'firstName': 'Quality',
            'lastName': 'Assurance',
            'email': 'qa@regent.edu',
            'password': hash_password('qa123'),
            'department': 'quality_assurance',
            'role': 'admin',
            'isActive': True,
            'createdAt': datetime.utcnow(),
            'updatedAt': datetime.utcnow()
        },
        {
            'firstName': 'President',
            'lastName': 'Office',
            'email': 'president@regent.edu',
            'password': hash_password('president123'),
            'department': 'presidency',
            'role': 'admin',
            'isActive': True,
            'createdAt': datetime.utcnow(),
            'updatedAt': datetime.utcnow()
        },
        {
            'firstName': 'VP',
            'lastName': 'Academics',
            'email': 'vpacademics@regent.edu',
            'password': hash_password('vpacademics123'),
            'department': 'vp_academics',
            'role': 'admin',
            'isActive': True,
            'createdAt': datetime.utcnow(),
            'updatedAt': datetime.utcnow()
        },
        {
            'firstName': 'Super',
            'lastName': 'Admin',
            'email': 'superadmin@regent.edu',
            'password': hash_password('superadmin123'),
            'department': 'admin',
            'role': 'super_admin',
            'isActive': True,
            'createdAt': datetime.utcnow(),
            'updatedAt': datetime.utcnow()
        }
    ]
    
    # Clear existing users
    db.users.delete_many({})
    
    # Insert new users
    result = db.users.insert_many(users)
    
    print(f"✅ Created {len(result.inserted_ids)} users successfully!")
    print("\nLogin credentials:")
    print("-" * 50)
    for user in users:
        print(f"Department: {user['department']}")
        print(f"Email: {user['email']}")
        print(f"Password: {user['email'].split('@')[0]}123")
        print("-" * 50)

if __name__ == '__main__':
    try:
        create_users()
        print("\n✅ Database seeding completed successfully!")
    except Exception as e:
        print(f"\n❌ Error: {e}")
    finally:
        client.close()

File: server/seed.js
// Database seeder for creating sample users
// Run with: node server/seed.js

import mongoose from 'mongoose';
import bcrypt from 'bcryptjs';
import dotenv from 'dotenv';

dotenv.config();

const MONGODB_URI = process.env.MONGODB_URI || 'mongodb://localhost:27017/regent_dms';

// User Schema (simplified for seeding)
const userSchema = new mongoose.Schema({
  firstName: String,
  lastName: String,
  email: String,
  password: String,
  department: String,
  role: String,
  isActive: Boolean,
});

const User = mongoose.model('User', userSchema);

const users = [
  {
    firstName: 'Registry',
    lastName: 'Admin',
    email: 'registry@regent.edu',
    password: 'registry123',
    department: 'registry',
    role: 'admin',
    isActive: true,
  },
  {
    firstName: 'Accounts',
    lastName: 'Admin',
    email: 'accounts@regent.edu',
    password: 'accounts123',
    department: 'accounts',
    role: 'admin',
    isActive: true,
  },
  {
    firstName: 'Quality',
    lastName: 'Assurance',
    email: 'qa@regent.edu',
    password: 'qa123',
    department: 'quality_assurance',
    role: 'admin',
    isActive: true,
  },
  {
    firstName: 'President',
    lastName: 'Office',
    email: 'president@regent.edu',
    password: 'president123',
    department: 'presidency',
    role: 'admin',
    isActive: true,
  },
  {
    firstName: 'VP',
    lastName: 'Academics',
    email: 'vpacademics@regent.edu',
    password: 'vpacademics123',
    department: 'vp_academics',
    role: 'admin',
    isActive: true,
  },
  {
    firstName: 'Super',
    lastName: 'Admin',
    email: 'superadmin@regent.edu',
    password: 'superadmin123',
    department: 'admin',
    role: 'super_admin',
    isActive: true,
  },
];

async function seedDatabase() {
  try {
    // Connect to MongoDB
    await mongoose.connect(MONGODB_URI);
    console.log('✅ Connected to MongoDB');

    // Clear existing users
    await User.deleteMany({});
    console.log('🗑️  Cleared existing users');

    // Hash passwords and create users
    const hashedUsers = await Promise.all(
      users.map(async (user) => {
        const salt = await bcrypt.genSalt(10);
        const hashedPassword = await bcrypt.hash(user.password, salt);
        return {
          ...user,
          password: hashedPassword,
        };
      })
    );

    // Insert users
    await User.insertMany(hashedUsers);
    console.log(`✅ Created ${hashedUsers.length} users successfully!`);

    console.log('\n📋 Login Credentials:');
    console.log('-'.repeat(50));
    users.forEach((user) => {
      console.log(`Department: ${user.department}`);
      console.log(`Email: ${user.email}`);
      console.log(`Password: ${user.password}`);
      console.log('-'.repeat(50));
    });

    console.log('\n✅ Database seeding completed successfully!');
    process.exit(0);
  } catch (error) {
    console.error('❌ Error:', error.message);
    process.exit(1);
  } finally {
    await mongoose.connection.close();
    console.log('\n👋 Database connection closed');
  }
}

seedDatabase();
